client/client.go

```diff
@@ -48,6 +48,23 @@ var DefaultTransport CancelableTransport = &http.Transport{
 	TLSHandshakeTimeout: 10 * time.Second,
 }
 
+type EndpointSelectionMode int
+
+const (
+	// EndpointSelectionDefault is to pick an endpoint in a random manner.
+	EndpointSelectionDefault EndpointSelectionMode = iota
+
+	// EndpointSelectionPrioritizeLeader is to prioritize leader for reducing needless
+	// forward between follower and leader.
+	//
+	// This mode should be used with Client.AutoSync().
+	EndpointSelectionPrioritizeLeader
+
+	// EndpointSelectionPreference is to manually picks an endpoint
+	// with using Config.PreferenceIndex as an index.
+	EndpointSelectionPreference
```

Comment:
Yea, we should keep the switch structure. I just want to remove `EndpointSelectionPreference` feature for now. I mentioned the idea as an example that we might want this option to be extensive. So we should make it an enum rather than just a bool. But I am not convinced we need this now and this is exactly what we need. I need to think it more.